### ISSUE TITLE
How to QuickCheck DB-related code?

### ISSUE THREAD

**Issue by saurabhnanda** [nick8325/quickcheck#106]
Already asked this on [Stackoverflow](http://stackoverflow.com/questions/38644779/how-to-use-quickcheck-to-test-database-related-functions) AND [Reddit](https://www.reddit.com/r/haskell/comments/4vfqqi/how_to_use_quickcheck_to_test_database_related/), but no one seems to know.

I've been trying to solve the problem for the past 3 days now, but no matter what I try, the types don't compose elegantly.

**Comment by reiddraper** [comment]
@saurabhnanda have you looked at `Test.QuickCheck.Monadic.monadicIO`? Here's a little example:

```haskell
example :: Integer -> Property
example x =
    _ <- $ do
        print x
        -- this is all in IO, so you could connect to a database,
        -- and do whatever you want
    assert True
```

**Comment by saurabhnanda** [comment]
@reiddraper do take a look at http://stackoverflow.com/a/38684717/534481 ? Isn't that a better way?

**Comment by reiddraper** [comment]
@saurabhnanda yeah, looks like that should work fine, too. I was only intending to show a basic example of performing `IO` during a test.
